sklearn/cluster/hierarchical.py

```diff
@@ -814,7 +834,7 @@ def fit(self, X, y=None):
         compute_full_tree = self.compute_full_tree
         if self.connectivity is None:
             compute_full_tree = True
-        if compute_full_tree == 'auto':
+        if compute_full_tree == 'auto' and self.distance_threshold is None:
```

Comment:
I think we need to remove this and explicitly force compute_full_tree to be True if distance_threshold is passed